### ISSUE TITLE
rpi gpio input node error

### ISSUE THREAD

**Issue by DrLucyRogers** [node-red/node-red#1139]
Hello,
Node-RED v0.15.2 
rpi gpio input node - Pin 40 has a switch attached. The oscilloscope says the switch is switching, but sometimes Node-RED is not picking it up. 
I also sporadically get "key 'node-red: (en-GB)' returned an object instead of a string. error message under the node. I can't force it to bring that message up, but it misses the swtiches after about 8 times, then misses for a while.
This is happening when I move the SD card and use it on either a Pi zero or Pi3.
I am also an mp3123 exec node (USB audio adapter), rpi-neopixels node and two other rpi-gpio output nodes.

Any ideas please?

Here's the messy flow: 

[{"id":"9e63265c.0356a8","type":"rpi-neopixels","z":"98348193.4b728","name":"","pixels":"16","bgnd":"black","fgnd":"black","wipe":"1","mode":"pixelsneedle","rgb":"rgb","x":680,"y":260,"wires":[]},{"id":"ce33eef7.dcaf1","type":"inject","z":"98348193.4b728","name":"","topic":"","payload":"0","payloadType":"num","repeat":"","crontab":"","once":false,"x":90,"y":20,"wires":[["c19f388c.2be958","7c5051a2.2a414"]]},{"id":"ef52105a.ba4548","type":"delay","z":"98348193.4b728","name":"","pauseType":"delay","timeout":"5","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"x":610,"y":140,"wires":[["ab0307cb.1c101"]]},{"id":"ab0307cb.1c101","type":"change","z":"98348193.4b728","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"Black","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":670,"y":200,"wires":[["9e63265c.0356a8"]]},{"id":"c19f388c.2be958","type":"function","z":"98348193.4b728","name":"Increment from 1 to N","func":"\nmsg.payload = parseInt(msg.payload) + 1;\n\nif (msg.payload <= 16)\n    return msg;\n","outputs":1,"noerr":0,"x":270,"y":80,"wires":[["f56f2323.290f38","7a2ec84e.c3a6f8"]]},{"id":"f56f2323.290f38","type":"function","z":"98348193.4b728","name":"Add White","func":"msg.payload = msg.payload + \",white\";\nreturn msg;","outputs":1,"noerr":0,"x":410,"y":140,"wires":[["9e63265c.0356a8","efa9fbce.ece5f","ef52105a.ba4548","11178056.85ddd"]]},{"id":"efa9fbce.ece5f","type":"debug","z":"98348193.4b728","name":"","active":true,"console":"false","complete":"false","x":600,"y":20,"wires":[]},{"id":"7a2ec84e.c3a6f8","type":"delay","z":"98348193.4b728","name":"","pauseType":"random","timeout":"200","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"250","randomLast":"500","randomUnits":"milliseconds","drop":false,"x":260,"y":20,"wires":[["c19f388c.2be958"]]},{"id":"11178056.85ddd","type":"exec","z":"98348193.4b728","command":"mpg123 Crackermulti2sec.mp3","addpay":true,"append":"","useSpawn":"","timer":"","name":"","x":680,"y":60,"wires":[[],[],[]]},{"id":"dd286876.f021d8","type":"rpi-gpio in","z":"98348193.4b728","name":"Switch","pin":"40","intype":"up","debounce":"50","read":false,"x":76,"y":303,"wires":[["160a10b3.03edcf"]]},{"id":"160a10b3.03edcf","type":"rbe","z":"98348193.4b728","name":"","func":"rbe","gap":"","start":"","inout":"out","x":90,"y":180,"wires":[["c3aaaf77.1c09b"]]},{"id":"a2dfbf86.8bdc3","type":"debug","z":"98348193.4b728","name":"","active":true,"console":"false","complete":"false","x":180,"y":520,"wires":[]},{"id":"c3aaaf77.1c09b","type":"change","z":"98348193.4b728","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"0","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":130,"y":120,"wires":[["a2dfbf86.8bdc3","c19f388c.2be958","7c5051a2.2a414"]]},{"id":"94d9fcef.9d6fb","type":"rpi-gpio out","z":"98348193.4b728","name":"","pin":"37","set":true,"level":"0","out":"out","x":720,"y":420,"wires":[]},{"id":"d2bf71f6.71c2e","type":"rpi-gpio out","z":"98348193.4b728","name":"","pin":"35","set":true,"level":"0","out":"out","x":760,"y":660,"wires":[]},{"id":"a9c2a8de.36f168","type":"change","z":"98348193.4b728","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"1","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":390,"y":280,"wires":[["d2bf71f6.71c2e","b16a7196.384c3","4571e972.fe6b28"]]},{"id":"4571e972.fe6b28","type":"delay","z":"98348193.4b728","name":"","pauseType":"delay","timeout":"8","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"x":400,"y":400,"wires":[["7139bbdc.2a49e4"]]},{"id":"b16a7196.384c3","type":"delay","z":"98348193.4b728","name":"","pauseType":"delay","timeout":"1","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"x":640,"y":320,"wires":[["94d9fcef.9d6fb"]]},{"id":"7139bbdc.2a49e4","type":"change","z":"98348193.4b728","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"0","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":470,"y":540,"wires":[["d2bf71f6.71c2e","94d9fcef.9d6fb"]]},{"id":"2970e9e6.59e836","type":"inject","z":"98348193.4b728","name":"","topic":"","payload":"1","payloadType":"str","repeat":"","crontab":"","once":false,"x":550,"y":680,"wires":[["94d9fcef.9d6fb","d2bf71f6.71c2e"]]},{"id":"d62fedb6.40c89","type":"inject","z":"98348193.4b728","name":"","topic":"","payload":"0","payloadType":"str","repeat":"","crontab":"","once":false,"x":520,"y":740,"wires":[["d2bf71f6.71c2e","94d9fcef.9d6fb"]]},{"id":"7c5051a2.2a414","type":"function","z":"98348193.4b728","name":"","func":"\nreturn msg;","outputs":1,"noerr":0,"x":370,"y":200,"wires":[["a9c2a8de.36f168"]]}]

Thanks Lucy

**Comment by DrLucyRogers** [comment]
Update:

Pi3. Node-RED v0.15.2

I have removed everything from the flow. It now just has an Rpi GPIO input node.

<img width="910" alt="screen shot 2017-02-05 at 21 33 22" src="https://cloud.githubusercontent.com/assets/3532238/22630089/e58ca40c-ebea-11e6-8580-d3098a24ae6b.png">

Node-RED is not recognising every switch. I have swapped the physical switch with a different one and the problem is the same. 

I don't think it is to do with the timing - it happens even if I wait a second between switches.
